src/test/java/lambda/part1/exercise/Exercise2.java

```diff
@@ -0,0 +1,105 @@
+package lambda.part1.exercise;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.StringJoiner;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.emptyString;
+import static org.hamcrest.Matchers.is;
+
+@SuppressWarnings({"FieldCanBeLocal", "ConstantConditions", "unused"})
+class Exercise2 {
+
+    @FunctionalInterface
+    private interface Multiplier<T> {
+
+        T multiply(T value, int multiplier);
+
+        default T twice(T t) {
+            return multiply(t, 2);
+        }
+    }
+
+    @Test
+    void implementsIntegerMultiplierUsingAnonymousClass() {
+        Multiplier<Integer> multiplier = null;
+
+        multiplier = new Multiplier<Integer>() {
+            @Override
+            public Integer multiply(Integer value, int multiplier) {
+                return value*multiplier;
+            }
+        };
+        testIntegerMultiplier(multiplier);
+    }
+
+    @Test
+    void implementsMultiplierUsingStatementLambda() {
+        Multiplier<Integer> multiplier = null;
+
+        multiplier = (i1, i2) -> {return i1*i2;};
+
+        testIntegerMultiplier(multiplier);
+    }
+
+    @Test
+    void implementsIntegerMultiplierUsingExpressionLambda() {
+        Multiplier<Integer> multiplier = null;
+
+        multiplier = (i1, i2) -> i1*i2;
+
+        testIntegerMultiplier(multiplier);
```

Comment:
I think you can inline multiplier

Reply:
@Ilya21 Yes, it looks better, thanks. Fixed)